client/state/posts/test/reducer.js

```diff
@@ -366,4 +370,105 @@ describe( 'reducer', () => {
 			expect( state ).to.eql( {} );
 		} );
 	} );
+
+	describe( '#siteRequests()', () => {
+		it( 'should default to an empty object', () => {
+			const state = siteRequests( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should map site ID, post ID to true value if request in progress', () => {
+			const state = siteRequests( undefined, {
+				type: POST_REQUEST,
+				siteId: 2916284,
+				postId: 841
+			} );
+
+			expect( state ).to.eql( {
+				2916284: {
+					841: true
+				}
+			} );
+		} );
+
+		it( 'should accumulate mappings', () => {
+			const state = siteRequests( Object.freeze( {
```

Comment:
>I think you need to deepfreeze to really verify, as the thing being mutated is a level down

Ooh, wow, you're totally right, and there are existing reducers which are breaking because of the failure of `Object.assign` to deeply clone objects too. The [`siteQueries` reducer](https://github.com/Automattic/wp-calypso/blob/b29d2b1c31642e8f34a1b7f2491401af5d35b62b/client/state/posts/reducer.js#L80-L104), for example, suffers from an issue similar to http://jsbin.com/puyuhepiju/edit?html,js,console .

And before any mention of Immutable.js is made, the use of `deepFreeze` in tests is effective in ensuring that these are properly handled :wink: